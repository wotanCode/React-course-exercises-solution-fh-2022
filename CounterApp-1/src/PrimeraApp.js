//import React, { Fragment } from "react";
import React from "react";
import PropTypes from "prop-types";

//FC funcional components
const PrimeraApp = ( {saludo, subtitulo } ) => {

  //const saludo = 'hola mundo'; //Valido
  // const saludo = 123; //Valido
  // const saludo = 123.2; //Valido
  // const saludo = true; //inValido
  // const saludo = [1,2,3,4,5] //Valido
  // const saludo = {
  //   nombre: 'hola',
  //   edad: 31,
  // }; //valido pero solo si usamos JSON.Stringify

  console.log(saludo)

  return (
    <>
      {/* <pre> { JSON.stringify(saludo, null, 3) } </pre> */}
      <h1>{saludo}</h1>
      <h2>{subtitulo}</h2>
    </>
  );
};

PrimeraApp.protoType = {
  saludo: PropTypes.string.isRequired,
}

PrimeraApp.defaultProps = {
  subtitulo: 'Soy un subtitulo',
}

export default PrimeraApp;