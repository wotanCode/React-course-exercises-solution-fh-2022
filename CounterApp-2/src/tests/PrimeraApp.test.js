import React from "react";
//import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme'
import "@testing-library/jest-dom"


describe('Pruebas en <Priemrapp />', () => {
  //test('Debe de mostrar el mensaje "Hola soy pedrito"', () =>{
    //EN eto tampoco funcionaria en enzy
    // const saludo = "Hola soy pedrito";
    // const wrapper = render(<PrimeraApp saludo={saludo} />)

    //esto hubiera sido otra forma de hacerlo
    //const {getByText} = render(<PrimeraApp saludo={saludo} />)
    //Abajo hubiera quedado
    //expect( getByText(saludo) )
    //Esto ya no funciona en enzy
    //expect( wrapper.getByText(saludo) ).toBeInTheDocument();

    test('Debe de mostrar <PrimeraApp/> correctamente"', () =>{
    const saludo = "Hola soy pedrito"
    const wrapper = shallow(<PrimeraApp saludo= {saludo}/>)

    expect( wrapper ).toMatchSnapshot();
  })

    test('Debe de mostrar el subtitulo enviado por props)', () => {
      const saludo = "Hola soy pedrito"
      const subtitulo ="Soy un subtitulo"
      const wrapper = shallow(
        <PrimeraApp
          saludo = {saludo}
          subtitulo = {subtitulo}
        />
      );

      const textoParrafo = wrapper.find('p').text();
      expect(textoParrafo).toBe(subtitulo);

    })
})