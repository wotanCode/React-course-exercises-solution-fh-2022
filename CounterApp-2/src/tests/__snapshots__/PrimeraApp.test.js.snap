// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Priemrapp /> Debe de mostrar <PrimeraApp/> correctamente" 1`] = `
<Fragment>
  <h1>
    Hola soy pedrito
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
